hydromt/_utils/caching.py

```diff
@@ -63,54 +112,101 @@ def _cache_vrt_tiles(
-    # TODO check if this is the optimal xml parser
     with open(vrt_destination_path, "r") as f:
-        ds = xd.parse(f.read())["VRTDataset"]
-
-    def intersects(
-        source: Dict[str, Dict[str, float]], affine: Affine, bbox: List[float]
-    ):
-        """Check whether source intersects with bbox."""
-        names = ["@xOff", "@yOff", "@xSize", "@ySize"]
-        x0, y0, dx, dy = [float(source["DstRect"][k]) for k in names]
-        xs, ys = affine * (np.array([x0, x0 + dx]), np.array([y0, y0 + dy]))
-        return (
-            max(xs) < bbox[0]
-            or max(ys) < bbox[1]
-            or min(xs) > bbox[2]
-            or min(ys) > bbox[3]
-        )
+        root = ET.fromstring(f.read())
 
     # get vrt transform and crs
-    transform = Affine.from_gdal(*literal_eval(ds["GeoTransform"]))
-    srs = ds["SRS"]["#text"] if isinstance(ds["SRS"], dict) else ds["SRS"]
-    crs = CRS.from_string(srs)
+    geotransform_el: Optional[ET.Element] = root.find("GeoTransform")
+    if geotransform_el is None:
+        raise ValueError(f"No GeoTransform found in: {vrt_uri}")
+    transform = Affine.from_gdal(*literal_eval(geotransform_el.text.strip()))
+
+    srs_el: Optional[ET.Element] = root.find("SRS")
+    if srs_el is None:
+        raise ValueError(f"No SRS info found at: {vrt_uri}")
+
+    crs: CRS = CRS.from_string(srs_el.text)
     # get geometry bbox in vrt crs
     if geom is not None:
         if crs != geom.crs:
             geom = geom.to_crs(crs)
         bbox = geom.total_bounds
     # support multiple type of sources in vrt
-    source_name = [k for k in ds["VRTRasterBand"].keys() if k.endswith("Source")][0]
+    band_name: Optional[ET.Element] = root.find("VRTRasterBand")
+    if band_name is None:
+        raise ValueError(f"Could not find VRTRasterBand in: {vrt_uri}")
+    try:
+        source_name: str = next(
+            filter(lambda k: k.tag.endswith("Source"), band_name.iter())
+        ).tag
+    except StopIteration:
+        raise ValueError(f"No Source information found at: {vrt_uri}")
+
     # loop through files in VRT and check if in bbox
     paths = []
-    for source in ds["VRTRasterBand"][source_name]:
-        if geom is None or intersects(source, transform, bbox):
-            path = source["SourceFilename"]["#text"]
-            dst = os.path.join(cache_dir, path)
-            src = os.path.join(vrt_root, path)
-            if not isfile(dst):
-                paths.append((src, dst))
-    # TODO multi thread download
+    for source in band_name.findall(source_name):
+        if geom is None or _overlaps(source, transform, bbox):
+            vrt_ref_el: Optional[ET.Element] = source.find("SourceFilename")
+            if vrt_ref_el is None:
+                raise ValueError(f"Could not find Source File in vrt: {vrt_uri}.")
+            vrt_ref: str = vrt_ref_el.text.replace("\\", os.sep)  # dewindowsify
+            vrt_relative: int = int(vrt_ref_el.get("relativeToVRT"))  # 0 or 1
```

Comment:
why int and not bool?

Reply:
The source format gives either 0 or 1, I like to represent it that way.